lib/install/deps.js

```diff
@@ -689,7 +689,6 @@ var findRequirement = exports.findRequirement = function (tree, name, requested,
     return null
   }
   if (tree.isTop) return null
-  if (!preserveSymlinks() && /^[.][.][\\/]/.test(path.relative(tree.parent.realpath, tree.realpath))) return null
```

Comment:
As you might imagine, this code was added for a reason and just removing it will introduce a new class of bugs.

In particular, this is would break how we resolve dependencies inside of symlinks, making npm think the dependency is resolvable even when it isn't. A solution to this bug will have to come some other way.

Reply:
I figured that it would break something, the tests that were added didn't pick that up btw.

Reply:
I saw that, which is surprising and definitely does call for a look.